api-call.py

```diff
@@ -1,29 +1,322 @@
+
+        text = json.dumps(data, indent=4)
+        return data
+
+    def show_tasks(self) -> None:
+        """Show tasks to choose as a Menu
+        """
+        api_options = ['Quit',
+                    'Beavers Bus',
+                    'Terms',
+                    'Search Text Books of a Term by a custom/any Date',
+                    'Get Details of a Vehicle and its Stops given a specified Route, with ETA and current destination']
         
-    api_call =  get_api_data("https://api.oregonstate.edu/v1/beaverbus/routes", parameters, header)
+        print('Which API data do you want to check?')
+        for i, option in enumerate(api_options):
+            print(f'{i}. {option}')
+
+    def get_user_choice(self) -> str:
+        """Get user input from menu
+
+        :returns: The user choice
+        """
+        return input('Enter your choice (0 or 1 or 2 or 3 or 4): ')
+
+    def get_url(self, choice: str) -> Dict[str, Any]:
+        """Get URL for API
+
+        :param choice: user choice input
+        :returns: API URL Object after user choice
+        """
+
+        access_token = self.get_access_token()
+        header = {'Content-Type': 'application/x-www-form-urlencoded;charset=utf-8', 'Authorization': access_token}
+        url_obj = {}
+        if choice == '0':
+            return 'Exiting...'
+        elif choice == '1':
+            url_obj = {'url': self.config['apiUrls']['beaver_bus'],
+                    'parameters': {},
+                    'header': header}
+        elif choice == '2':
+            url_obj = {'url': self.config['apiUrls']['terms'],
+                    'parameters': {},
+                    'header': header}
+        elif choice == '3':
+            user_date = input('Enter Date (yyyy-mm-dd): ')
+            url_obj = {'url_terms': self.config['apiUrls']['terms'],
+                    'url_textbooks': self.config['apiUrls']['textbooks'],
+                    'parameters': {'date': user_date},
+                    'header': header}
+        elif choice == '4':
+            route_id = input('Enter Route Number: ')
+            url_obj = {'url_routes': self.config['apiUrls']['routes'],
+                    'url_arrivals': self.config['apiUrls']['arrivals'],
+                    'url_vehicles': self.config['apiUrls']['vehicles'],
+                    'route_id': route_id,  'parameters': {},
+                    'header': header}
+        else:
+            print('Invalid choice. Please try again.')
+
+        return url_obj
+
+    def get_text_books_with_term_date(self, url_obj: Dict[str, Any]) -> None:
+        """Get text books based on specified term and date
+        """
+
+        terms_data_response = self.get_api_data(
+            url_obj['url_terms'],
+            url_obj['parameters'],
+            url_obj['header']
+        )
+        if terms_data_response:
+            terms_data = terms_data_response.get('data', [])
+            if terms_data:
```

Comment:
What if `terms_data` is empty? What a user should expect to see when using this script?